app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/signup/SignupViewModel.kt

```diff
@@ -1,6 +1,65 @@
 package com.github.alfabravo2013.readyforexams.presentation.signup
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.alfabravo2013.readyforexams.R
+import com.github.alfabravo2013.readyforexams.repository.AuthenticationResult
+import com.github.alfabravo2013.readyforexams.repository.LoginRepositoryImpl
+import com.github.alfabravo2013.readyforexams.util.SingleLiveEvent
+import com.github.alfabravo2013.readyforexams.util.isInvalidEmail
+import com.github.alfabravo2013.readyforexams.util.isInvalidPassword
+import kotlinx.coroutines.launch
 
 class SignupViewModel : ViewModel() {
+    private val loginRepository = LoginRepositoryImpl
+
+    private val _onEvent = SingleLiveEvent<OnEvent>()
+    val onEvent: SingleLiveEvent<OnEvent> get() = _onEvent
+
+    var signUpSuccess = false
```

Comment:
If you need this you should move it to a localDataSource. If you plan to do it later you can leave it for now. Also booleans should start with `is or has or should etc` for example: isSignUpSuccessful